instrumentation/akka-http-10.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkahttp/server/AkkaHttpServerInstrumentationModule.java

```diff
@@ -66,28 +72,32 @@ public AsyncWrapper(
 
     @Override
     public Future<HttpResponse> apply(HttpRequest request) {
-      Context ctx = tracer().startSpan(request, request, null, "akka.request");
-      try (Scope ignored = ctx.makeCurrent()) {
+      Context parentContext = currentContext();
+      if (!instrumenter().shouldStart(parentContext, request)) {
+        return userHandler.apply(request);
+      }
+      Context context = instrumenter().start(parentContext, request);
+      try (Scope ignored = context.makeCurrent()) {
         return userHandler
             .apply(request)
             .transform(
                 new AbstractFunction1<HttpResponse, HttpResponse>() {
                   @Override
                   public HttpResponse apply(HttpResponse response) {
-                    tracer().end(ctx, response);
+                    instrumenter().end(context, request, response, null);
                     return response;
                   }
                 },
                 new AbstractFunction1<Throwable, Throwable>() {
                   @Override
                   public Throwable apply(Throwable t) {
-                    tracer().endExceptionally(ctx, t);
+                    instrumenter().end(context, request, errorResponse(), t);
```

Comment:
This is a bit strange. Here and on line 56, I needed to return a default error response so that the tests associated the attributes expected by the test with the span.

Reply:
Hmm, which test cases failed? Was it because of status code?

Reply:
The `HttpServerTest."test exception"` cases fail for `AkkaHttpServerInstrumentationTestSync` and `AkkaHttpServerInstrumentationTestAsync` if these `errorResponse()` are missing.

Reply:
can you open an issue for this? probably we should revisit

the old tracer had this code which has its own problems:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b0430f2b79d2ec8c1805c0889bdbc06e5e5f0af1/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java#L137-L138

Reply:
Servlet instrumenter has the same issue https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b0430f2b79d2ec8c1805c0889bdbc06e5e5f0af1/instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletHttpAttributesExtractor.java#L79-L93
If the actual response status is set outside the scope of our integration we'll need to widen our integration so that we can capture the actual response code or predict what it is going to be.
Just using `errorResponse()` would be incorrect in case it is possible to send response status and fail with exception after that. Servlet instrumentation uses `isResponseCommitted` to make sure that we don't replace the status that was already sent to client with `500`.

Reply:
thx! i'm in favor of merging this as-is 👍